src/xaa.ts

```diff
@@ -142,9 +148,9 @@ export class TimeoutRunner {
    * @param tasks - Promise or function that returns Promise, or array of them.
    * @returns Promise to wait for tasks to complete, or timeout error.
    */
-  async run(tasks: Promise<any> | Function | Array<Function | Promise<any>>): Promise<any> {
-    const process = (x: any) => (typeof x === "function" ? x() : x);
-    let arrTasks: Promise<any[]>;
+  async run(tasks) {
+    const process = async (x: any) => (typeof x === "function" ? x() : x);
+    let arrTasks: Promise<any>;
```

Comment:
should this not be `Promise<T>[]`? if the internals are of the tasks are mixed types, unfortunately i think we'd have do use the arg matching, multiple arity T inference boilerplate for various "overloads" if we wanted nice inference behavior. (e.g. `(tasks: [], ...): Promise<[]>`, `(tasks: [t0: T0], ...): Promise<[Unpack<T0>, Unpack<T1>]>`, `(tasks: [t0: T0, t1: T1])`, ...)` etc

Reply:
Unfortunately they are of rather mixed types, Joel did a rough sketch (these should be accurate I think)

```
type TimeoutTaskFunc<T> = () => Promise<T>;
type TimeoutTask<T> = Promise<T> | TimeoutTaskFunc;
type TimeoutTasks<T> = TimeoutTask<T> | TimeoutTask<T>[];
```

then finally
`async run<Something>(tasks: Something)`

but I'll have to look later into how we can improve it.